### PR TITLE
User feedback fixes to D7YIGPositionCalibration

### LINKED ISSUE #30861: User feedback to D7YIGPositionCalibration
Following the user feedback certain modifications to the algorithm and its documentation are needed:

1. Fix unwanted renaming of the workspace provided through `InputWorkspace` property.
2. Consider addding loading feature for the InputWorkspace so that it doesn't need to be pre-loaded each time.
3. Include Chi^2 value in the fit output information.
4. Fix naming of intermediate workspaces in case the optional `FitOutputWorkspace` property is not specified.

### PR DESCRIPTION
**Pull request by darominski**
This PR fixes the issues and new feature requests brought up by D7 users:

- [x] Unwanted renaming of the workspace provided through InputWorkspace property
- [x] Include Chi^2 value in the fit output information
- [x] Fix naming of intermediate workspaces in case the optional FitOutputWorkspace property is not specified.

The loading feature along with the InputWorkspace is currently not added, as after fixing the renaming issue, the work so that it doesn't need to be pre-loaded each time.
   
The documentation for the PolarizedDiffractionILL has been updated with a more comprehensive example and explanation of the possibility to save time by caching the inputs.

**To test:**

1. Make sure you have ILL D7 UnitTest data in your user search directories
2. Load 402652_403041.nxs or other nxs file that contains underscore, e.g. using the example from [documentation](https://docs.mantidproject.org/nightly/algorithms/D7YIGPositionCalibration-v1.html)
3. Run the following script in mantid. You may need to modify the InputWorkspace and ApproximateWavelength relevant to your case:
```
from mantid.simpleapi import *

approximate_wavelength = '3.1' # Angstrom
D7YIGPositionCalibration(
		InputWorkspace='402652_403041',		
		ApproximateWavelength=approximate_wavelength,
                YIGPeaksFile='D7_YIG_peaks.xml',
                MinimalDistanceBetweenPeaks=1.5,
		BraggPeakWidth=1.5,
		BankOffsets=[0,0,0],
                MaskedBinsRange=[-50, -25, 15],
		FittingMethod='None',
		ClearCache=False,
                FitOutputWorkspace='fitting_test')
```

4. Verify that:
a) calibration runs without any issues,
b) there are no issues if you comment out the YIGPeaksFile property,
c) the input workspace is not be renamed to anything else,
d) if you change the FitOutputWorkspace property to empty string, the intermediate workspaces should contain 'calibration' string. 

Fixes #30861. 

*This does not require release notes* because these changes are small corrections that restore the expected behaviour.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by gvardany** [comment]
I got an error for b), when removing the yig peaks file:
```Property "YIGPEAKSFILE" is not set to a valid value: "File "D7_YIG_peaks.xml" not found".
Property "YIGPEAKSFILE" is not set to a valid value: "File "D7_YIG_peaks.xml" not found".
RuntimeError: ['YIGPeaksFile'] argument(s) not supplied to D7YIGPositionCalibration
  File "<string>", line 15, in <module>
  File "/Users/vardanyan/mantid/Framework/PythonInterface/mantid/simpleapi.py", line 1046, in __call__
    _check_mandatory_args(name, algm, e, *args, **kwargs)
  File "/Users/vardanyan/mantid/Framework/PythonInterface/mantid/simpleapi.py", line 709, in _check_mandatory_args
    raise RuntimeError("%s argument(s) not supplied to %s" % (missing_arg_list, algorithm))```

I see you are taking the cost function value from the table which is the chi-squared. More informative is the chi-squared / number of degrees of freedom. But this should be changed in the fit itself to write that value instead in the table.

File: Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/D7YIGPositionCalibration.py [hunk 1]
```diff
@@ -53,6 +53,9 @@ def validateInputs(self):
         if self.getPropertyValue('FittingMethod') != 'None' and self.getProperty('CalibrationOutputFile').isDefault:
             issues['CalibrationOutputFile'] = 'The name of the calibration output needs to be defined'
 
+        if self.getPropertyValue('YIGPeaksFile') == '':
```
Comment (gvardany) [inline Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/D7YIGPositionCalibration.py:56]: Is this still needed?

Comment (darominski) [reply Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/D7YIGPositionCalibration.py]: Not at all, I've just removed it from the last commit.

File: docs/source/techniques/PolarisedDiffractionILL.rst [hunk 2]
```diff
@@ -87,26 +87,36 @@ The first step of working with D7 data is to ensure that there exist a proper ca
...
-   D7YIGPositionCalibration(Filenames='402652:403041', ApproximateWavelength=approximate_wavelength,
-                            YIGPeaksFile='D7_YIG_peaks.xml',
-                            MinimalDistanceBetweenPeaks=1.5, BankOffsets=[0,0,0],
-                            MaskedBinsRange=[-50, -25, 15], FittingMethod='None', ClearCache=False,
-                            FitOutputWorkspace='fitting_test')
+   D7YIGPositionCalibration(
+		Filenames='402652:403041',
+   #		InputWorkspace='conjoined_input_fitting_test',
+		ApproximateWavelength=approximate_wavelength,
+		YIGPeaksFile='D7_YIG_peaks.xml',
```
Comment (gvardany) [inline docs/source/techniques/PolarisedDiffractionILL.rst:112]: The default is now a full path, so what happens if the user gives the base name of the default file?

Comment (darominski) [reply docs/source/techniques/PolarisedDiffractionILL.rst:112]: That is processed without issues, just like when algorithm dialog was used with the previous solution. The search for the file is triggered correctly.

**Review by gvardany — APPROVED** [review]
Ok.

**Pull request merged**